### ISSUE TITLE
File.dirname normalization differs from MRI

### ISSUE THREAD

**Issue by rjnienaber** [jruby/jruby#2391]
From: http://jira.codehaus.org/browse/JRUBY-4863. Differs from MRI on JRuby 1.7.18 and JRuby 9000.

MRI strips additional leading slashes with File.dirname:

```
File.dirname("////foo/bar/baz") => /foo/bar (MRI)
File.dirname("////foo/bar/baz") => ////foo/bar (JRuby)
```

However, I'm not sure if this "normalization" is intentional in MRI or accidental, because it only seems to work on **leading** slashes in MRI:

```
File.dirname("////foo//bar/baz") => /foo//bar (MRI)
File.dirname("////foo//bar/baz") => ////foo//bar (JRuby)
```

I'd say the spec needs to be fleshed out. Either the path should be completely normalized, or not at all.

**Comment by rtyler** [comment]
I've reproduced this with the latest 9k and 1.7.21 FWIW

**Comment by enebo** [comment]
I fixed this some time during 9.1.x and we have EOLd 1.7.x.  Resolving.
